04-DFS&BFS/chaerin/1260_DFS와BFS.py

```diff
@@ -0,0 +1,53 @@
+from collections import deque
+
+
+def DFS(graph, root):
+    visited = []
+    stack = [root]
+
+    while stack:
+        n = stack.pop()
+        if n not in visited:
+            visited.append(n)
+            if n in graph:
+                temp = list(set(graph[n]) - set(visited))
+                temp.sort(reverse=True)
+                stack += temp
+
+    return " ".join(str(i) for i in visited)
+
+
+def BFS(graph, root):
+    visited = []
+    queue = deque([root])
+
+    while queue:
+        n = queue.popleft()
+        if n not in visited:
+            visited.append(n)
+            if n in graph:
+                temp = list(set(graph[n]) - set(visited))
+                temp.sort()
+                queue += temp
+    return " ".join(str(i) for i in visited)
+
+
+graph = {}
+n = input().split(' ')
+node, edge, start = [int(i) for i in n]
+for i in range(edge):
+    edge_info = input().split(' ')
+    n1, n2 = [int(j) for j in edge_info]
+    if n1 not in graph:
+        graph[n1] = [n2]
+    elif n2 not in graph[n1]:
+        graph[n1].append(n2)
+
+    if n2 not in graph:
+        graph[n2] = [n1]
+    elif n1 not in graph[n2]:
+        graph[n2].append(n1)
+# print(graph)
```

Comment:
이차원 배열로 간선 연결상태를 표현해도 좋을 것 같아요 !!

```suggestion
 N, M, V = map(int, input().split())  # 정점의 개수, 간선의 개수, 탐색 시작할 정점의 번호
 visit = [[0 for j in range(N)] for i in range(N)]  # 정점 간의 연결 상태를 확인하기 위함

for _ in range(M):  # 간선의 개수만큼 간선 정보를 받아와야 한다
    i, j = map(int, input().split())
    visit[i - 1][j - 1] = 1
    visit[j - 1][i - 1] = 1
```